### PR TITLE
[Gutenberg] Audio upload processor

### PR DESCRIPTION
**Pull request by etoledom**
`gutenberg` PR: https://github.com/WordPress/gutenberg/pull/27401
`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2854

This PR implements the Audio upload processor to be used together with the Audio block.

This PR also improves the file type filet for the on-device file picker: Now only file extensions allowed by the blog can be picked. Previously any video/image extensions could be picked for the Audio/Video blocks. This is also now used on the Audio block to avoid Apple-only audio filetypes (common on audio files recorded in iPhones).

**To test:**
Test running metro server on https://github.com/wordpress-mobile/gutenberg-mobile/pull/2854

- On a new post, add an Audio block.
- Add to it a relatively large Audio file.
- While the upload is happening, close the editor saving as a draft.
  - The post card will show an upload in progress.
- Wait for the upload to finish and the post to be saved remotely.
- Open the saved post.
- Switch to HTML mode.
  - Check that the Audio block id has a positive number.
  - Check that the `src` URL is a remote (non-local) URL.

cc @jd-alexander 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergAudioUploadProcessor.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import Foundation
+import Aztec
+
+class GutenbergAudioUploadProcessor: Processor {
+    private struct AudioBlockKeys {
+        static var name = "wp:audio"
```
Comment (guarani) [inline WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergAudioUploadProcessor.swift]:
```suggestion
        static let name = "wp:audio"
```

File: WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergAudioUploadProcessor.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        static var id = "id"
```
Comment (guarani) [inline WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergAudioUploadProcessor.swift]:
```suggestion
        static let id = "id"
```

File: WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergAudioUploadProcessor.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        static var src = "src"
```
Comment (guarani) [inline WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergAudioUploadProcessor.swift]:
```suggestion
        static let src = "src"
```

**Review by guarani — APPROVED** [review]
Looks good to me @etoledom! Left a few minor code comments.

**Comment by etoledom** [comment]
Thank you @Guarani 🙏 
Nice catch! Committed the suggestions 👍

**Pull request merged**